### PR TITLE
Feature/bus 912 no products

### PR DESCRIPTION
**Pull request by carojansing**
Adds a No Product Banner in Listing

### CODE REVIEW

File: components/Cms/Element/CmsElementProductListing.vue [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <script setup lang="ts">
 import type { CmsElementText } from '@shopware-pwa/composables-next';
+import SharedBanner from '~/components/shared/SharedBanner.vue';
```
Comment (mloeffle) [inline components/Cms/Element/CmsElementProductListing.vue:3]:
```suggestion
```

File: components/Cms/Element/CmsElementProductListing.vue [hunk 2]
```diff
@@ -54,7 +55,9 @@ const changePage = async (page: number) => {
 </script>
 
 <template>
+    <div v-if="getElements[0]">
     <div class="grid grid-cols-2 gap-6 md:grid-cols-3 lg:grid-cols-4">
```
Comment (mloeffle) [inline components/Cms/Element/CmsElementProductListing.vue:59]: i think this div should be indented, did you run `bun run lint`

File: components/Shared/SharedBanner.vue [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<script setup lang="ts">
+const props = withDefaults(
```
Comment (mloeffle) [inline components/Shared/SharedBanner.vue:2]:
```suggestion
withDefaults(
```

File: components/Shared/SharedBanner.vue [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        <div class="flex items-center justify-center bg-brand-primary-dark h-full w-10">
+        <div class="border-full flex items-center justify-center h-6 w-6 rounded-full border-2">
+        <FormKitIcon
+            icon="info"
+            class="h-3 w-3 block rounded-full "
+        />
+        </div>
+        </div>
+        <div class="pl-2 message text-brand-primary-dark">
+            {{ props.message }}
```
Comment (mloeffle) [inline components/Shared/SharedBanner.vue:24]:
```suggestion
            {{message }}
```

File: components/Cms/Element/CmsElementProductListing.vue [hunk 5]
```diff
@@ -54,22 +54,29 @@ const changePage = async (page: number) => {
...
-    </div>
+    <div v-if="getElements[0]">
+        <div class="grid grid-cols-2 gap-6 md:grid-cols-3 lg:grid-cols-4">
+            <template
+                v-for="product in getElements"
+                :key="product.id"
+            >
+                <ProductCard
+                    :product="product"
+                    :layout="element.translated.config.boxLayout.value"
```
Comment (aheartforspinach) [inline components/Cms/Element/CmsElementProductListing.vue:65]: could you adapt it like its mentioned here? https://github.com/basecom/Pond/pull/52#discussion_r1773364124

File: components/Cms/Element/CmsElementProductListing.vue [hunk 6]
```diff
@@ -54,22 +54,29 @@ const changePage = async (page: number) => {
...
+            :total="getTotal"
+            :items-per-page="getLimit"
+            :default-page="getCurrentPage"
+            @update-page="page => changePage(page)"
+        />
+    </div>
+    <SharedBanner
+        v-else
+        type="info"
+        message="No Products found"
```
Comment (aheartforspinach) [inline components/Cms/Element/CmsElementProductListing.vue:80]: please use i18n

File: components/Cms/Element/CmsElementProductListing.vue (hunk 5 above)
Comment (Marius-BSCM) [reply components/Cms/Element/CmsElementProductListing.vue]: I implemented this to the best of my knowledge. Did you mean it that way?

Comment (aheartforspinach) [reply components/Cms/Element/CmsElementProductListing.vue]: yeah exactly this way! thank you!

**Review by aheartforspinach — APPROVED** [review]

**Review by mloeffle — APPROVED** [review]

**Pull request merged**